Frameworks/CoreFoundation/Locale.subproj/CFCalendar.c

```diff
@@ -746,6 +746,12 @@ static CFRange __CFCalendarGetRangeOfUnit2(CFCalendarRef calendar, CFCalendarUni
     // WINOBJC: Add cases for weekOf units. ICU will handle them below.
     case kCFCalendarUnitWeekOfYear:
     case kCFCalendarUnitWeekOfMonth:
+    // WINOBJC: Asking for the range of WeekOfMonth in Year always returns 6.
```

Comment:
OK, I see the problem here. The calculate logic works by setting the starting date to the first moment in the larger unit, then advancing in increments of the smaller unit to find the min and max values. The problem is that this may not work correctly when the smaller unit doesn't divide the larger unit evenly.

For example, try getting the range of WeekOfMonth in Month for July 2016. The answer should be [1..6], but you will actually get [1..5].

This also explains why you sometimes get the wrong answer when asking for the range of WeekOfMonth in Year. It's not returning the number of weeks in the current month; it's advancing from Jan 1 week by week through the year, querying the WeekOfMonth each time.

The logic in calculate should be fixed.

---
In reply to: [74359393](https://github.com/Microsoft/WinObjC/pull/844#discussion_r74359393) [](ancestors = 74359393,74356593,74354002,74342094,74336418)